src/plugin/safe-empty-array-where-in/safe-empty-array-where-in.transformer.ts

```diff
@@ -0,0 +1,64 @@
+import { BinaryOperationNode } from '../../operation-node/binary-operation-node.js'
+import { OperationNode } from '../../operation-node/operation-node.js'
+import { OperationNodeTransformer } from '../../operation-node/operation-node-transformer.js'
+import { PrimitiveValueListNode } from '../../operation-node/primitive-value-list-node.js'
+import { WhereNode } from '../../operation-node/where-node.js'
+
+const IN_OPERATORS = ['in', 'not in'] as const
+
+interface InOperatorNode {
+  kind: 'OperatorNode'
+  operator: (typeof IN_OPERATORS)[number]
+}
+
+export class WithSafeArrayWhereInTransformer extends OperationNodeTransformer {
+  #BinaryOperationNode(node: OperationNode): node is BinaryOperationNode {
+    return node.kind === 'BinaryOperationNode'
+  }
+
+  #isPrimitiveValueListNode(
+    node: OperationNode,
+  ): node is PrimitiveValueListNode {
+    return node.kind === 'PrimitiveValueListNode'
+  }
+
+  #IsInOperatorNode(node: OperationNode): node is InOperatorNode {
+    return (
+      (node as InOperatorNode).kind === 'OperatorNode' &&
+      IN_OPERATORS.includes((node as InOperatorNode).operator)
+    )
+  }
+
+  protected transformWhere(node: WhereNode): WhereNode {
```

Comment:
what about `having`?

Reply:
Good point. With https://github.com/kysely-org/kysely/pull/925/commits/d0230ad19305046d8b06648dc13dc25e99368032 the transformation is now done when traversing a `BinaryOperationNode`, which covers this case.

Added tests [here](https://github.com/kysely-org/kysely/pull/925/commits/d0230ad19305046d8b06648dc13dc25e99368032)